src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHandlerManager.cs

```diff
@@ -11,12 +11,18 @@ namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
     [Export]
     internal class LanguageServiceHandlerManager
     {
+        private readonly ICommandLineParserService _commandLineParser;
+
         [ImportingConstructor]
-        public LanguageServiceHandlerManager(UnconfiguredProject project)
+        public LanguageServiceHandlerManager(UnconfiguredProject project, ICommandLineParserService commandLineParser)
         {
             Requires.NotNull(project, nameof(project));
+            Requires.NotNull(commandLineParser, nameof(commandLineParser));
+
+            _commandLineParser = commandLineParser;
 
             EvaluationHandlers = new OrderPrecedenceImportCollection<IEvaluationHandler>(projectCapabilityCheckProvider: project);
+            CommandLineHandlers = new OrderPrecedenceImportCollection<ICommandLineHandler>(projectCapabilityCheckProvider: project);
```

Comment:
Synced offline with Dave - while this is unsymmetric, not doing it this way adds a lot of complexity because in a future change these handlers won't be mef imported and will be created per context. In that world, having a separate component create a handler like the source handler which is both an evaluation and designtime one will complicate lifetime management.